Style images

Artemis21 opened the pull request:
Add styling for images to each of the themes.

Artemis21 committed:
Style images in hotdog theme.

Make them switch between red and yellow like the text.

Artemis21 committed:
Syle images in Window 95 theme.

Make them small and unsaturated, and expand them (and return the colour to normal) on hover.

Artemis21 committed:
Style images in Unicorn theme.

Make them over saturated and over bright, and move them to the opposite side of the page every 4s. Hover on an image for 5s to return it to normal.

Artemis21 committed:
Style images in Console theme.

Make them 30x30 pixel thumbnails that expand to a full screen modal when hovered.

Artemis21 committed:
Style images in Galaxy theme.

Blur in an out with an over the top border.

Artemis21 committed:
Add border image for Galaxy theme.

Artemis21 committed:
Make indentation consistent for CSS.

A5rocks commented on jolly-jellyfish/src/static/filters.svg:
```diff
@@ -0,0 +1,22 @@
+
+<svg xmlns="http://www.w3.org/2000/svg" version="1.1" xmlns:xlink="http://www.w3.org/1999/xlink" width="0" height="0">
+    <filter id="red">
+        <feColorMatrix type="matrix" values="
+            1   0.6 0.6 0 1
+            0.3 0   0   0 0
+            0.3 0   0   0 0
+            0   0   0   1 0
+        "/>
```
could you add a comment about what these values mean?

Artemis21 replied:
Added

A5rocks commented on jolly-jellyfish/src/static/filters.svg:
```diff
@@ -0,0 +1,22 @@
+        "/>
+    </filter>
+    <filter id="yellow">
+        <feColorMatrix type="matrix" values="
+            0.5  0.5 0.3 0 0.5
+            0.5  0.5 0.3 0 0.5
+            0.15 0   0   0 0
+            0    0   0   1 0
+        "/>
```
see above

Artemis21 replied:
see above

A5rocks commented on jolly-jellyfish/src/static/themes/console.css:
```diff
@@ -13,6 +13,43 @@ body {
+}
+
+.img-wrapper:hover::before {
+  content: "<";
```
not sure if I like this... maybe we could adjust the transparency (using some css transitions)? (of course, if it looks great ignore this)

Artemis21 replied:
That exists as a button to close the modal

A5rocks replied:
Oh okay, nice.

A5rocks commented on jolly-jellyfish/src/static/themes/unicorn.css:
```diff
@@ -7,17 +7,18 @@
-              url("/static/images/unicorn8.gif") right bottom / 33% 33% no-repeat,
-              url("/static/images/unicorn9.gif") left top / 33% 33% no-repeat,
-              linear-gradient(to right, red, orange, yellow, green, blue, indigo, violet) 100% 100% repeat;
+  background:
+    url("../images/unicorn0.gif") left top / 33% 33% no-repeat,
+    url("../images/unicorn1.gif") left center / 33% 33% no-repeat,
+    url("../images/unicorn2.gif") left bottom / 33% 33% no-repeat,
+    url("../images/unicorn3.gif") center top / 33% 33% no-repeat,
+    url("../images/unicorn4.gif") center center / 33% 33% no-repeat,
+    url("../images/unicorn5.gif") center bottom / 33% 33% no-repeat,
+    url("../images/unicorn6.gif") right top / 33% 33% no-repeat,
+    url("../images/unicorn7.gif") right center / 33% 33% no-repeat,
+    url("../images/unicorn8.gif") right bottom / 33% 33% no-repeat,
+    url("../images/unicorn9.gif") left top / 33% 33% no-repeat,
+    linear-gradient(to right, red, orange, yellow, green, blue, indigo, violet) 100% 100% repeat;
```
Not sure how much I like the small amount of indentation here... Maybe indent more? (there was another spot with this problem too... but I already passed it xd)

Artemis21 replied:
I will go back to the visual indent, sure

A5rocks commented on jolly-jellyfish/src/static/themes/win95.css:
```diff
@@ -3,6 +3,7 @@
  * A static file "/static/fonts/hellovetica.woff2" should exist,
  * but it will fall back on "News Cycle" from Google Fonts if not.
  * Static files "/static/images/win95-toolbar.png" and "/static/images/win95-close.png" should also exist.
+ * Requires that images be wrapped in divs.
```
Mind writing what class is required? (if one is... I didn't see any new things about divs being related to images in the file...)

Artemis21 replied:
This theme isn't picky about the class, but I do need to put that in the Console and Galaxy themes.

This theme does still require the divs though, to make 100% relative to the size of the image.

A5rocks commented on jolly-jellyfish/src/static/themes/win95.css:
```diff
@@ -28,6 +29,20 @@ body {
 
+img {
+  width: 100%;
+  height: 100px;
```
Should we really be using pixel measurements for something?

I'm not sure if it appears anywhere else though, so maybe we already are and I missed it...

Artemis21 replied:
There are quite a few things using pixels, I take your point for this one though, maybe I should change it to 2em or something.

A5rocks reviewed (commented):
Can't comment on the design much, just a couple nitpicks :)

Artemis21 committed:
Explain image wrapping requirements and uses.

Artemis21 committed:
Use a visual indent for background images.

Artemis21 committed:
Use an em measurement rather than a pixel measurement for initial image height in the Windows 95 theme.

Artemis21 committed:
Improve the red and yellow colour filter matrices.

Take what we do allow in non-target colours, from the non-target colours for clearer images.

Artemis21 committed:
Attempt to explain colour matrices.

A5rocks reviewed (approved):
LGTM from a cursory overview
